docs/proposals/MultiSelect/MultiSelect.md

```diff
@@ -0,0 +1,105 @@
+# `MultiSelect`
+
+The `MultiSelect` component gives our users the possibility to select multiple
+options from a menu.
+
+## Design Patterns
+
+The goal of the `MultiSelect` component is to allow a user to check multiple
+options within a list of items.
+
+A user can click on the element to expand it and display the multiple options.
+The element will display the selected options or the placeholder text if none
+are selected.
+
+Every time an option is checked or unchecked a behavior can be triggered using
+the onChange prop
+
+### How it might behave:
+
+- This multi-select dropdown matches the designs seen in the Figma here
+
+- Developer should have props for whether or not an “All” checkbox should show
+  up at the top, as well as the text that should display in this option
+
+  - The “All” checkbox should be indeterminate
+
+    - If it’s currently unchecked and the user checks it then all of the lower
+      options should be checked automatically
+
+      - If unchecked again after that then all of the lower options should be
+        unchecked automatically
+
+    - The “All” checkbox is automatically marked checked if the user manually
+      checks all of the options below it
+
+    - The “All” checkbox is automatically marked unchecked if the user manually
+      unchecks all of the options below it
+
+- The text that appears in the box before a user clicks on it should be either
```

Comment:
How would we handle a long list of selected options? Does the trigger element grow, or would it truncate?

Reply:
I think we could add a prop to fix the width but let the default behavior to adjust to its content, is that ok?

Reply:
For the QBO design specifically we will show max two options then truncate the third
![image](https://user-images.githubusercontent.com/27180625/194603052-a9a90fbf-7dc1-473c-88ef-51c207d3fc9f.png)

Reply:
OK so the behaviour is always to truncate?

Reply:
As for the current QBO design, yes. But looking at the Finsights design it show "2 selected", "3 selected". Maybe it could be a switch on/off feature?

Reply:
What if the content was just a "label"? Then the responsive behaviour is to always truncate overflow, but what's displayed in the element could be set by each team. 

So in your case...

`someFunctionToGetTheStatusesForTheLabel`

`label={ someFunctionToGetTheStatusesForTheLabel }`

And for the other usage

`label={ numberOfSelectedFilters + 'selected' }`

?

Reply:
yes! I believe this is a good approach to leave the decision of what should be the label to the developer with a label parameter. 

Thank you